snuba/query/validation/validators.py

```diff
@@ -95,10 +98,14 @@ class SubscriptionAllowedClausesValidator(QueryValidator):
     clauses are being used in the query, and that those clauses are in the correct structure.
     """
 
+    def __init__(self, entity: Optional["Entity"] = None) -> None:
+        self.entity = entity
```

Comment:
This is causing an import loop because entities are supposed to contain validators, not the other way around. However, the query knows which entity it is being called on, you can get it using this class: https://github.com/getsentry/snuba/blob/master/snuba/query/parser/validation/functions.py#L33

You can then do whatever check you need from there.